test/e2e/framework/networking_utils.go

```diff
@@ -269,8 +269,33 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
 func (config *NetworkingTestConfig) GetSelfURL(path string, expected string) {
 	cmd := fmt.Sprintf("curl -q -s --connect-timeout 1 http://localhost:10249%s", path)
 	By(fmt.Sprintf("Getting kube-proxy self URL %s", path))
-	stdout := RunHostCmdOrDie(config.Namespace, config.HostTestContainerPod.Name, cmd)
-	Expect(strings.Contains(stdout, expected)).To(BeTrue())
+
+	// These are arbitrary timeouts. The curl command should pass on first try,
+	// unless kubeproxy is starved/bootstrapping/restarting etc.
+	interval := 1 * time.Second
+	timeout := 30 * time.Second
+	podName := config.HostTestContainerPod.Name
+	var msg string
+	if pollErr := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		stdout, err := RunHostCmd(config.Namespace, podName, cmd)
+		if err != nil {
+			msg = fmt.Sprintf("failed executing cmd %v in %v/%v: %v", cmd, config.Namespace, podName, err)
+			Logf(msg)
```

Comment:
doesn't Logf do formatting as part of the function?

Logf(""failed executing cmd %v in %v/%v: %v", cmd, ...)